students/kahyee/Session04/mailroom2.py

```diff
@@ -0,0 +1,70 @@
+import os
+
+donation_dict = {
+"Bob": [1.00, 2.00],
+"Jon": [1.50, 100.00],
+"Sally": [1000.00], 
+"Barry": [50.00],
+"Ellen": [1.25]
+}
+
+def create_report():
+	print("Donor Name          | Total Given | Num Gifts | Average Gift")
+	print("------------------------------------------------------------")
+	for donor, amounts in donation_dict.items():
+		gift_count = str(len(donor) - 1)
+		gift_total = "%.2f" % sum(amounts)
+		gift_avg = "%.2f" % (sum(amounts) / len(donor))
+		print(donor \
+			+ " " * (20 - len(donor)) \
+			+ " $" + " " * (12 - len(gift_total)) + gift_total \
+			+ " " * 10 + str(gift_count) \
+			+ " $" + " " * (12 - len(gift_avg)) + gift_avg)
+
+
+def thank_you():
+    donor_name = input("Donor's Full Name: ")
+    exist = False
+    while donor_name.lower() == "list":
+        [print(donor) for donor in donation_dict.keys()]
+        donor_name = input("Donor's Full Name: ")
+    donation_amt = float(input("How much did they donate?"))
+    for donor, amounts in donation_dict.items():
+        if donor == donor_name:
+            amounts.append(donation_amt)
```

Comment:
Huh, I didn't realize you could add the donation to the amounts variable, and it would add it to the dictionary, but it does make sense! Interesting find. Setdefault would also be handy here.